File: src/dotnet/APIView/ClientSPA/src/app/_models/createUpdateCommentModel.ts
import { CommentType } from "./commentItemModel";

export class CreateCommentModel
{
    reviewId: string = '';
    elementId: string = '';
    commentText: string = '';
    commentType: CommentType | null = null;
    apiRevisionId: string = '';
    sampleRevisionId: string = '';
    resolutionLocked: boolean = false;
}

export class UpdateCommentModel
{
    reviewId: string = '';
    commentId: string = '';
    commentText: string = '';
}

File: src/dotnet/APIView/APIViewWeb/LeanModels/CreateUpdateCommentModel.cs
namespace APIViewWeb.LeanModels
{
    public class CreateCommentModel
    {
        public string ReviewId { get; set; }
        public string ElementId { get; set; }
        public string CommentText { get; set; }
        public CommentType CommentType { get; set; }
        public string ApiRevisionId { get; set; }
        public string SampleRevisionId { get; set; }
        public bool ResolutionLocked { get; set; }
    }

    public class UpdateCommentModel
    {
        public string ReviewId { get; set; }
        public string CommentId { get; set; }
        public string CommentText { get; set; }
    }
}
